lib/src/Pos/Diffusion/Full/Types.hs

```diff
@@ -28,6 +29,7 @@ type DiffusionWorkMode m
     = ( WithLogger m
       , CanJsonLog m
       , MonadMockable m
+      , MonadKnownPeers m
```

Comment:
This must go. We don't need it. You can update the known peers by grabbing the `OutboundQueue` directly. Full diffusion layer has one in scope.

Reply:
The `dnsSubscriptionWorker`is also used where we don't have access to the `OutboundQ` ([here](https://github.com/input-output-hk/cardano-sl/blob/develop/lib/src/Pos/Worker.hs#L55)).  I will add the `updatePeersBucket` function to the type signature so we can pass it as the first argument either from the `MonadKnownPeers` constraint or by using `updatePeersBucket` function directly when the `OutboundQ` is in scope).